### ISSUE TITLE
Send an email to new users on registration

### ISSUE THREAD

**Issue by KawaMood** [meeting-room-booking-system/mrbs-code#3756]
Hi, 

I tried to explore the code in `edit_user.php` to add my own layer in it, but I genuinely don't understand how it works.
My aim is to let a new user know that their account has been created by sending them an email. So basically, get the email you set in the email field when you create a new user, and send them an email.

Is there any hook for a new user registration I can exploit? How can I get the post data from the submit to retrieve the email address?

Thank you in advance.

**Comment by campbell-m** [comment]
Do you mean edit_user.php from the roles branch, or edit_users.php from the main branch?  They work slightly differently.

**Comment by KawaMood** [comment]
I mean when you create a new user as an administrator (so from the main user list), automatically send them an email to warn them their account has been created. I finally managed to do something within the `edit_user_handler.php` but it is very junky.
Is there maybe a proper way to do so?

**Comment by KawaMood** [comment]
I didn't actually. Does the system change how mail are sent? As I use sendmail in local with Wamp for my tests, but it doesn't work especially with MRBS for some reason I got the error : `Failed to connect to mailserver at "localhost" port 25, verify your "SMTP" and "smtp_port" setting in php.ini or use ini_set() in ........`

Also the edit_user_handler.php seems to be called twice, so it send two times my logs, and I think I will have an error with the mail being either sent two times or MRBS will say headers have already been defined or so...

If you have a valid solution I'm interested. :(

**Comment by campbell-m** [comment]
I'm not sure why you are getting edit_user_handler.php being called twice.  It is normally only called once.

You should use MRBS's mail system by using `MailQueue::add()`.   Add the mail message to the queue after the line `$user->save();` in edit_user_handler.php.

**Comment by jberanek** [comment]
Pretty sure you're not using your local sendmail app, and instead have configured MRBS to use SMTP, which is very different indeed.

**Comment by KawaMood** [comment]
Hi, 
Thank you both for your quick answers.

I didn't get the time to dive into it further until tonight, I finally managed to solve my issue, it was my fault. I fixed an issue on my side with my sendmail app, so I think jbernaek was right on this.
I may also have poorly manipulated the tool concerning the double-execution of the handler the last time, as from the tests I made today, it execute only once as expected. Sorry for that.

I will mark this thread as resolved!
